File: 1-two-sum/1-two-sum.scala
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        var map: Map[Int, Int] = Map()
        var res: Array[Int] = null
        val len = nums.length
        
        for(i <- 0 to len - 1) {
            val dif = target - nums(i)
            if (map.contains(dif)) {
                res = Array(map(dif), i)
            }
            map += (nums(i) -> i)
        }
        
        res
    }
}

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var map = new Map()
    var res = []
    
    nums.forEach((n, i) => {
        var dif = target - n
        if (map.has(dif)) {
            res = [map.get(dif), i]
        } else {
            map.set(n, i)
        }
    })
    
    return res
};